pyanaconda/payload/rpmostreepayload.py

```diff
@@ -239,46 +239,73 @@ def install(self):
 
         mainctx.pop_thread_default()
 
+    # Internal API for setting up bind mounts between the physical root and
+    # sysroot, also ensures we track them in self._internal_mounts so we can
+    # cleanly unmount them.
+    def _setupInternalBindmount(self, src, dest=None,
+                                src_physical=True,
+                                bind_ro=False,
+                                recurse=True):
+        # Default to the same basename
+        if dest is None:
+            dest = src
+        # Almost all of our mounts go from physical to sysroot
+        if src_physical:
+            src = iutil.getTargetPhysicalRoot() + src
+        else:
+            src = iutil.getSysroot() + src
+        # Canonicalize dest to the full path
+        dest = iutil.getSysroot() + dest
+        if bind_ro:
+            self._safeExecWithRedirect("mount",
+                                       ["--bind", src, src])
+            self._safeExecWithRedirect("mount",
+                                       ["--bind", "-o", "remount,ro", src, src])
```

Comment:
That's how it works, see `man mount`.

>        mount(8) since v2.27 allows to change the mount options by passing the relevant options along with --bind.  For example:
>
>              mount --bind,ro foo foo
>
>       This feature is not supported by the Linux kernel; it is implemented in userspace by an additional mount(2) remounting system call.  This solution is not atomic.
>
>       The alternative (classic) way to create a read-only bind mount is to use the remount operation, for example:
>
>              mount --bind olddir newdir
>              mount -o remount,bind,ro olddir newdir

(We could use the `mount --bind,ro` feature but I'd need to undo it when backporting to RHEL)